src/main/java/reposense/authorship/analyzer/AnnotatorAnalyzer.java

```diff
@@ -23,43 +24,45 @@
+        Optional<Author> currentAnnotatedAuthor = Optional.empty();
         Path filePath = Paths.get(fileInfo.getPath());
         for (LineInfo lineInfo : fileInfo.getLines()) {
             if (lineInfo.getContent().contains(AUTHOR_TAG)) {
-                Author newAuthor = findAuthorInLine(lineInfo.getContent(), authorAliasMap);
+                Optional<Author> newAnnotatedAuthor = findAuthorInLine(lineInfo.getContent(), authorAliasMap);
 
-                if (newAuthor.equals(Author.UNKNOWN_AUTHOR)) {
-                    //end of an author tag should belong to this author too.
-                    lineInfo.setAuthor(currentAuthor);
-                } else if (newAuthor.getIgnoreGlobMatcher().matches(filePath)) {
-                    newAuthor = Author.UNKNOWN_AUTHOR;
+                if (!newAnnotatedAuthor.isPresent()) {
+                    // end of an author tag should belong to the current author too.
+                    // if an end author tag was used without a corresponding starting tag, attribute the
+                    // line to UNKNOWN_AUTHOR
+                    lineInfo.setAuthor(currentAnnotatedAuthor.orElse(Author.UNKNOWN_AUTHOR));
+                } else if (newAnnotatedAuthor.get().getIgnoreGlobMatcher().matches(filePath)) {
+                    newAnnotatedAuthor = Optional.empty();
                 }
 
                 //set a new author
-                currentAuthor = newAuthor;
-            }
-            if (!currentAuthor.equals(Author.UNKNOWN_AUTHOR)) {
-                lineInfo.setAuthor(currentAuthor);
+                currentAnnotatedAuthor = newAnnotatedAuthor;
             }
+            currentAnnotatedAuthor.ifPresent(lineInfo::setAuthor);
         }
     }
 
     /**
      * Extracts the author name from the given {@code line}, finds the corresponding {@code Author}
-     * in {@code authorAliasMap},and returns the result.
-     * @return {@code Author#UNKNOWN_AUTHOR} if no matching {@code Author} is found.
+     * in {@code authorAliasMap}, and returns this {@code Author} stored in an {@code Optional}.
+     * @return {@code Optional.of(Author#UNKNOWN_AUTHOR)} if no matching {@code Author} is found,
+     *         {@code Optional.empty()} if an end author tag is used (i.e. "@@author"),
+     *         and {@code Optional.empty()} if the extracted author name is too short.
      */
-    private static Author findAuthorInLine(String line, Map<String, Author> authorAliasMap) {
+    private static Optional<Author> findAuthorInLine(String line, Map<String, Author> authorAliasMap) {
         try {
             String[] split = line.split(AUTHOR_TAG);
             String name = extractAuthorName(split[1]);
             if (name == null) {
-                return Author.UNKNOWN_AUTHOR;
+                return Optional.empty();
             }
-            return authorAliasMap.getOrDefault(name, Author.UNKNOWN_AUTHOR);
+            return Optional.of(authorAliasMap.getOrDefault(name, Author.UNKNOWN_AUTHOR));
```

Comment:
@jylee-git, @fzdy1914, I realised there's another aspect that will be good to discuss now.

When the `AUTHOR_NAME` in `@@author AUTHOR_NAME` is not provided in our `AuthorConfiguration`, `AnnotatorAnalyzer` will attribute the subsequent code block to `UNKNOWN_AUTHOR`.

This fixes the *current* bug in the disowning mechanism. Users will be able to disown code.

However, as mentioned in #1082, *in future* we would like to have a ramp chart for the "virtual authors" who have been credited with all the disowned code. This will allow us to view all the disowned code via the `Code Panel` button.

It is currently undecided whether we want a single ramp chart for all the virtual authors (`some-randomuser1`, 
`some-randomuser2`, etc) or separate ramp charts for each virtual author.  Prof mentioned that we can adopt the solution that requires less development effort.

Given our future goals, perhaps I should adjust this PR to prevent alot of rework in future?
If we want a **separate** ramp chart for each virtual author, then we cannot attribute all the disowned code blocks to `UNKNOWN_AUTHOR`. We would need to create an `Author` instance for each virtual author
(e.g. `some-randomuser1`, `some-randomuser2`, etc).

Reply:
My view is that we should create a single ramp chart for all virtual authors, as this should involve less development work.

Reply:
As discussed in #1082, it would be preferred to have separate ramp charts for virtual authors. I'll hence be tweaking the implementation of this bugfix.